sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -37,8 +37,10 @@ object InMemoryRelation {
       batchSize: Int,
       storageLevel: StorageLevel,
       child: SparkPlan,
-      tableName: Option[String]): InMemoryRelation =
-    new InMemoryRelation(child.output, useCompression, batchSize, storageLevel, child, tableName)()
+      tableName: Option[String],
+      stats: Option[Statistics]): InMemoryRelation =
```

Comment:
also name it `statsOfPlanToCache `

Reply:
and use `None` as default value to avoid updating the tests

Reply:
for using None, looks like scala does not allow me to do that 

with the earlier change for exactly this purpose in https://github.com/apache/spark/pull/19864/commits/4c347014f32aa3c3088f8a18b08786fb074a1790

it failed to compile https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/84499/console


Looks like Scala compiler does not consider whether the default parameter is from the same or different function parameter lists